src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleAutoRod.kt

```diff
@@ -0,0 +1,193 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LiquidBounce. If not, see <https://www.gnu.org/licenses/>.
+ */
+package net.ccbluex.liquidbounce.features.module.modules.combat
+
+import net.ccbluex.liquidbounce.config.types.ToggleableConfigurable
+import net.ccbluex.liquidbounce.event.tickHandler
+import net.ccbluex.liquidbounce.features.module.Category
+import net.ccbluex.liquidbounce.features.module.ClientModule
+import net.ccbluex.liquidbounce.features.module.modules.combat.ModuleAutoRod.facingEnemy.ignoreOnEnemyLowHealth
+import net.ccbluex.liquidbounce.features.module.modules.combat.killaura.ModuleKillAura
+import net.ccbluex.liquidbounce.features.module.modules.combat.killaura.features.KillAuraAutoBlock
+import net.ccbluex.liquidbounce.utils.client.Chronometer
+import net.ccbluex.liquidbounce.utils.combat.PriorityEnum
+import net.ccbluex.liquidbounce.utils.combat.TargetTracker
+import net.ccbluex.liquidbounce.utils.entity.getActualHealth
+import net.ccbluex.liquidbounce.utils.inventory.Slots
+import net.minecraft.client.option.KeyBinding
+import net.minecraft.entity.Entity
+import net.minecraft.entity.LivingEntity
+import net.minecraft.item.Items
+import net.minecraft.text.Text
+
+
+object ModuleAutoRod : ClientModule("AutoRod", Category.COMBAT) {
+
+    object facingEnemy : ToggleableConfigurable(this, "FacingEnemy", true) {
+
+        object ignoreOnEnemyLowHealth : ToggleableConfigurable(this, "IgnoreOnEnemyLowHealth", true) {
+            val enemyHealthThreshold by int(
+                "EnemyHealthThreshold", 5, 1..20
+            )
+            val playerHealthThreshold by int("PlayerHealthThreshold", 5, 1..20)
+        }
+    }
+
+    init {
+        tree(facingEnemy)
+        tree(ignoreOnEnemyLowHealth)
+    }
+
+    private val enemiesNearby by int("EnemiesNearby", 1, 1..5)
+    private val targetTracker = tree(TargetTracker(PriorityEnum.DIRECTION))
+    private val escapeHealthThreshold by int("EscapeHealthThreshold", 10, 1..20)
+    private val pushDelay by int("PushDelay", 100, 50..1000)
+    private val pullbackDelay by int("PullbackDelay", 500, 50..1000)
+    private val onUsingItem by boolean("OnUsingItem", false)
+    private val pushTimer = Chronometer()
+    private val rodPullTimer = Chronometer()
+    private var rodInUse = false
+    private var switchBack: Int? = -1
+    private var range by float("Range", 5f, 1f..10f)
+
+    override fun disable() {
+        KeyBinding.setKeyPressed(mc.options.useKey.boundKey, false)
+    }
+
+    val tickHandler = tickHandler {
+        // Check if player is using rod
+        val usingRod =
+            (mc.player?.isUsingItem == true && mc.player?.mainHandStack?.item == Items.FISHING_ROD) || rodInUse
```

Comment:
`mc.player.isUsingItem` can be replaced with `player.isUsingItem`
```suggestion
            (player.isUsingItem && player.mainHandStack.item == Items.FISHING_ROD) || rodInUse
```